### PR TITLE
Fix inconsistent in scalr_varialbe migration

### PR DESCRIPTION
**Pull request by vmotso**
### Changelog
* [ ] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [ ] I have updated resource/data source documentation in [docs](../docs)

### State
* [ ] My changes affect the state
* [ ] I have included a state migration and a unit test for it

### CODE REVIEW

**Comment by vmotso** [comment]
The customer has the error:
```
Error reading configuration of workspace ws-t3ij9m991fg: Error unpacking workspace ID: invalid workspace ID format: ws-t3ij9m9491fg (expected <ENVIRONMENT_ID>/<WORKSPACE>)
```
It is due to inconsistent state:
```json
  "resources": [
    {
      "type": "scalr_variable",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "workspace_id": "ws-<some-id>"
          },
        }
      ]
    }
  ]
```
"workspace_id": "ws-some-id" and "schema_version": 0 cannot be in the same state in normal flow.

I fixed it.

**But**

In customer state we see inconsistent scalr_workspace state:
```json
        {
          "schema_version": 0,
          "attributes": {
            "auto_apply": false,
            "environment_id": "env-t1amxzczzxt9prn5g0",
            "id": "ws-t3iicxvx1eqq748",
            "name": "ami-test-stage",
            "operations": true,
            "queue_all_runs": true,
            "ssh_key_id": null,
            "terraform_version": "0.12.28",
            "vcs_repo": [
              {
                "branch": "stage",
                "identifier": "terraform/scalr-iacp/apps/ami-test",
                "oauth_token_id": "vcs-t0uqcxvul3eks4g",
                "path": ""
              }
            ],
            "working_directory": ""
          }
```
"environment_id": "env-some-id" and "schema_version": 0 cannot be in the same state in normal flow.

And if the customer will trigger a run after this fix he will get the error:
```
 Workspace "ami-test-stage" is already in use for organization "env-t2fxcbcvakircvat8".
```
Due to inconsistent state terraform want to create a new workspace.

I suppose problems with migration were the result of [release](https://github.com/Scalr/terraform-provider-scalr/pull/11) shifts and [migration](https://github.com/Scalr/terraform-provider-scalr/pull/14) but I'm not sure I failed to reproduce it.

---

Summing up - after this fix the customer will have the error `Workspace is already in use ...` there are two ways: bad, and very bad.

very bad:  for scalr_workspace, manually change in the statefile schema_version: 0 -> 3.
bad: before the run delete these workspaces.

---

We will take a closer look at the migration mechanism... because it works not 100% as we expect.

**Comment by vmotso** [comment]
Update: Lee suggested a [solution](https://github.com/Scalr/terraform-provider-scalr/pull/48/commits/075b21229d36b7a673a19bbf7f7f96b135b25fea) that will help not to recreate the workspace.
